app/gateway/postgres/repositories/deliveries_delete.go

```diff
@@ -0,0 +1,25 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+)
+
+func (r *DeliveriesRepository) Delete(ctx context.Context, id int32) error {
+	const (
+		operation = "Repository.DeliveriesRepository.Delete"
+	)
+
+	query := `
+		UPDATE deliveries
+		SET deleted_at = current_timestamp
+		WHERE id = $1;
+	`
+
+	_, err := r.Client.Pool.Exec(ctx, query, id)
+	if err != nil {
+		return fmt.Errorf("%s: %w", operation, err)
+	}
+
+	return nil
```

Comment:
```suggestion
	query := `
		UPDATE deliveries
		SET deleted_at = COALESCE(deleted_at, current_timestamp)
		WHERE id = $1;
	`

	res, err := r.Client.Pool.Exec(ctx, query, id)
	if err != nil {
		return fmt.Errorf("%s: %w", operation, err)
	}
	
	if res.RowsAffected() == 0 {
		return fmt.Errorf("%s: %s", operation, "no rows in result set")
	}

	return nil
```

Dessa forma já validamos se o delivery não existir sem o GetByID e evitamos atualizar o deleted_at se já tiver sido deletado alguma vez